src/api/entities/Identity/TokenPermissions.ts

```diff
@@ -0,0 +1,295 @@
+
+        return isSto || isAsset;
+      });
+    }
+
+    /*
+     * Authorized:
+     *   - corporateAction
+     *   - corporateBallot
+     *   - capitalDistribution
+     */
+    if (group.isPolymeshV1Caa) {
+      return transactions.every(tag =>
+        [
+          ModuleName.CorporateAction,
+          ModuleName.CorporateBallot,
+          ModuleName.CapitalDistribution,
+        ].includes(tag.split('.')[0] as ModuleName)
+      );
+    }
+
+    const groupId = group.asCustom;
+
+    const groupPermissionsOption = await externalAgents.groupPermissions(rawTicker, groupId);
+
+    const permissions = extrinsicPermissionsToTransactionPermissions(
+      groupPermissionsOption.unwrap()
+    );
+
+    if (permissions === null) {
+      return true;
+    }
+
+    const { type, exceptions, values } = permissions;
+
+    /*
+     * if type is include:
+     *  all passed tags are in the values array AND are not in the exceptions array (isInValues && !isInExceptions)
+     * if type is exclude:
+     *  all passed tags are not in the values array OR are in the exceptions array (!isInValues || isInExceptions)
+     */
+    const isPresent = (tag: TxTag, flipResult: boolean) => {
+      const isInValues = values.some(value => isModuleOrTagMatch(value, tag));
+      const isInExceptions = !!exceptions?.includes(tag);
+
+      const result = isInValues && !isInExceptions;
+
+      return flipResult ? result : !result;
+    };
+
+    const isInclude = type === PermissionType.Include;
+
+    return transactions.every(tag => isPresent(tag, isInclude));
+  }
+
+  /**
+   * Retrieve this Identity's Permission Group for a specific Security Token
+   */
+  public async getGroup({
+    token,
+  }: {
+    token: string | SecurityToken;
+  }): Promise<CustomPermissionGroup | KnownPermissionGroup> {
+    const {
+      context: {
+        polymeshApi: {
+          query: { externalAgents },
+        },
+      },
+      context,
+      parent: { did },
+    } = this;
+
+    const ticker = getTicker(token);
+
+    const rawTicker = stringToTicker(ticker, context);
+    const rawIdentityId = stringToIdentityId(did, context);
+
+    const rawGroupPermissions = await externalAgents.groupOfAgent(rawTicker, rawIdentityId);
+
+    if (rawGroupPermissions.isNone) {
+      throw new PolymeshError({
+        code: ErrorCode.ValidationError,
+        message: 'This Identity is no longer an Agent for this Security Token',
+      });
+    }
+    const agentGroup = rawGroupPermissions.unwrap();
+
+    return agentGroupToPermissionGroup(agentGroup, ticker, context);
+  }
+
+  /**
+   * Retrieve the identifier data (block number, date and event index) of the event that was emitted when this Identity was granted
+   *   permissions over a specific Security Token
+   *
+   * @note uses the middleware
+   * @note there is a possibility that the data is not ready by the time it is requested. In that case, `null` is returned
+   */
+  public async grantedAt({
```

Comment:
I think it should be renamed to agentAddedAt or something similar to keep it similar to the event we are fetching.